cmd/activator/main.go

```diff
@@ -127,10 +127,14 @@ func main() {
 	a := activator.NewRevisionActivator(kubeClient, servingClient, logger)
 	a = activator.NewDedupingActivator(a)
 
-	// Retry on 503's for up to 60 seconds. The reason is there is
-	// a small delay for k8s to include the ready IP in service.
+	// Retry on 502's, 503's and 504's for up to 60 seconds.
+	// For 503's the reason is there is a small delay for k8s to include the ready IP in service.
 	// https://github.com/knative/serving/issues/660#issuecomment-384062553
-	shouldRetry := activatorutil.RetryStatus(http.StatusServiceUnavailable)
+	// For 502's and 504's the reason is that some apps (like java) may take time to server requests
+	// even after the container is running
+	// https://github.com/knative/eventing/issues/671#issuecomment-446711743
+	shouldRetry := activatorutil.RetryStatus(http.StatusServiceUnavailable,
```

Comment:
These should just be separate `RetryCond`s passed to `NewRetryRoundTripper`, which already supports this.